src/Adapter/Product/Update/ProductStockUpdater.php

```diff
@@ -0,0 +1,183 @@
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Pack\Exception\ProductPackConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Exception\ProductStockConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Exception\ProductStockException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Exception\StockAvailableNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Stock\StockManager;
+use Product;
+use StockAvailable;
+
+/**
+ * Updates settings related to Product stock
+ */
+class ProductStockUpdater
+{
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var StockManager
+     */
+    private $stockManager;
+
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @var StockAvailableRepository
+     */
+    private $stockAvailableRepository;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     * @param StockManager $stockManager
+     * @param ProductRepository $productRepository
+     * @param StockAvailableRepository $stockAvailableRepository
+     */
+    public function __construct(
+        ConfigurationInterface $configuration,
+        StockManager $stockManager,
+        ProductRepository $productRepository,
+        StockAvailableRepository $stockAvailableRepository
+    ) {
+        $this->configuration = $configuration;
+        $this->stockManager = $stockManager;
+        $this->productRepository = $productRepository;
+        $this->stockAvailableRepository = $stockAvailableRepository;
+    }
+
+    /**
+     * @param Product $product
+     * @param array $propertiesToUpdate
+     * @param bool $addMovement
+     *
+     * @throws CoreException
+     * @throws ProductConstraintException
+     * @throws ProductPackConstraintException
+     * @throws ProductStockConstraintException
+     * @throws ProductStockException
+     */
+    public function update(Product $product, array $propertiesToUpdate, bool $addMovement = true): void
+    {
+        $stockAvailable = $this->getStockAvailable($product);
+        $this->productRepository->partialUpdate($product, $propertiesToUpdate, CannotUpdateProductException::FAILED_UPDATE_STOCK);
+
+        // It is very important to update StockAvailable after product, because the validation is performed in ProductRepository::partialUpdate
+        $this->updateStockAvailable($product, $stockAvailable, $propertiesToUpdate, $addMovement);
+
+        $advancedStockEnabled = (bool) $this->configuration->get('PS_ADVANCED_STOCK_MANAGEMENT');
```

Comment:
Should we pass to constructor the whole `$configuration` service or just this boolean?